core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -349,9 +352,10 @@ private Boolean resolveUseFipsFromDefaultProvider(SdkClientConfiguration config)
     /**
      * Resolve the credentials that should be used based on the customer's configuration.
      */
-    private AwsCredentialsProvider resolveCredentials(SdkClientConfiguration config) {
-        return config.option(AwsClientOption.CREDENTIALS_PROVIDER) != null
-               ? config.option(AwsClientOption.CREDENTIALS_PROVIDER)
+    private IdentityProvider<? extends AwsCredentialsIdentity> resolveCredentialsIdentityProvider(SdkClientConfiguration config) {
+        // Note, that CREDENTIALS_PROVIDER is never set. It is replaced with CREDENTIALS_IDENTITY_PROVIDER, so just check that
```

Comment:
These types of comments are probably better placed on the option itself than the place we're using it.